src/patch/createPatch.ts

```diff
@@ -41,8 +41,14 @@ function _diff(from: any, to: any, startingPath?: JsonPointer): Operation[] {
 	return operations;
 }
 
-export function diff<T, U>(from: T, to: U): Patch<T, U> {
-	return <Patch<T, U>> createPatch(_diff(from, to));
+export function diff<T>(to: T): Patch<any, T>;
+export function diff<T, U>(from: T, to: U): Patch<T, U>;
+export function diff(from: any, to?: any) {
+	if (!to) {
+		to = from;
+		from = {};
+	}
+	return createPatch(_diff(from, to));
```

Comment:
I don't actually disagree that logically `from` Is normally first but both the API and the implementation confused me with swapping the order of the args and then depending on the order swapping the allocation in the impl.